### ISSUE TITLE
BUG: Value become 0 while exporting (to_excel) the str data

### ISSUE THREAD

**Issue by handy1988** [pandas-dev/pandas#39950]
- [X ] I have checked that this issue has not already been reported.

- [X ] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example
```python
import csv
import pandas as pd

output_csv = pd.read_csv(r"D:\test\output.csv")
output_csv.to_excel(r"D:\test\output.xlsx", index=False)

```

#### Problem description
Hi All, 
I found some issue while exporting "str" data from some column to excel, And I noticed this issue occurred only after pandas version 1.1.5 and above. And attached my sample of data for your reference. Kindly help to check. Thanks 
[files.zip](https://github.com/pandas-dev/pandas/files/6017378/files.zip)

#### output.csv
Col1,Col2
a1,50060e8007123400
a2,50060e8007123400
a3,50060e8007123400
a4,50060e8007123402
a5,50060e8007123402
a6,50060e8007123402
a7,50060e8007123402
a8,50060e8007123402
a9,50060e8007123404
a10,50060e8007123404

#### output.xlsx -> Current output
Col1 | Col2
-- | --
a1 | 0
a2 | 0
a3 | 0
a4 | 0
a5 | 0
a6 | 0
a7 | 0
a8 | 0
a9 | 0
a10 | 0

#### Expected Output
Col1 | Col2
-- | --
a1 | 50060e8007123400
a2 | 50060e8007123400
a3 | 50060e8007123400
a4 | 50060e8007123402
a5 | 50060e8007123402
a6 | 50060e8007123402
a7 | 50060e8007123402
a8 | 50060e8007123402
a9 | 50060e8007123404
a10 | 50060e8007123404

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 7d32926db8f7541c356066dcadabf854487738de
python           : 3.9.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : English_United States.1252

pandas           : 1.2.2
numpy            : 1.20.1
pytz             : 2021.1
setuptools       : 51.1.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.20.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.6
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

**Comment by MarcoGorelli** [comment]
Hi @handy1988 

Could you please put a descriptive title, and also show the output of `output_csv` as it's just 10 rows?

**Comment by handy1988** [comment]
Hi Marco,

Sorry, I forgot to update the title, now edited already with output of csv. Thanks

**Comment by phofl** [comment]
Hm when executing the following:

```
s = """
Col1,Col2
a1,50060e8007123400
a2,50060e8007123400
a3,50060e8007123400
a4,50060e8007123402
a5,50060e8007123402
a6,50060e8007123402
a7,50060e8007123402
a8,50060e8007123402
a9,50060e8007123404
a10,50060e8007123404
"""

output_csv = pd.read_csv(StringIO(s))
```

I get 
```
  Col1  Col2
0   a1   0.0
1   a2   0.0
2   a3   0.0
3   a4   0.0
4   a5   0.0
5   a6   0.0
6   a7   0.0
7   a8   0.0
8   a9   0.0
9  a10   0.0
```

Looks to me as the issue is in read_csv. Also your column is no text data, the *e* is interpreted as exponent.

**Comment by phofl** [comment]
You can do
```
output_csv = pd.read_csv(StringIO(s), dtype="object")
output_csv.to_excel(r"output.xlsx", index=False)
```

to get your result

**Comment by handy1988** [comment]
HI phofl,

Thanks for your reply, initially I thought this is a bug since I never faced this issue on previous version. I've tested it and it's working fine. Thanks again for your help👍

**Comment by phofl** [comment]
The conversion to 0 is in the same place as #38794, so closing in favor of that issue
